### ISSUE TITLE
PYI034 false positive when `Self` is imported from a local re-export module

### ISSUE THREAD

**Issue by jab** [astral-sh/ruff#24559]
### Summary

`PYI034` flags an `__ior__` method whose return type is already `Self`, if `Self` is imported from a local re-export module rather than directly from `typing` or `typing_extensions`.

### Reproducer

Given two files:

**`_mytyping.py`** (re-exports `Self`):
```python
import sys

if sys.version_info >= (3, 11):
    from typing import Self as Self
else:
    from typing_extensions import Self as Self
```

**`example.py`**:
```python
from __future__ import annotations

from collections.abc import Mapping
from typing import TypeVar

from _mytyping import Self

KT = TypeVar("KT")
VT = TypeVar("VT")


class Foo(Mapping[KT, VT]):
    def __getitem__(self, key: KT) -> VT: ...
    def __iter__(self): ...
    def __len__(self): ...

    def __ior__(self, other: Mapping[KT, VT]) -> Self:
        return self
```

```
$ ruff check example.py --select PYI034
error[PYI034]: `__ior__` methods in classes like `Foo` usually return `self` at runtime
  --> example.py:18:9
   |
16 |     def __len__(self): ...
17 |
18 |     def __ior__(self, other: Mapping[KT, VT]) -> Self:
   |         ^^^^^^^
19 |         return self
   |
help: Use `Self` as return type

Found 1 error.
```

### Expected behavior

No error — the return type is already `Self`.

### Comparison

If `Self` is imported directly from `typing` instead of from the local re-export module, `PYI034` correctly recognizes it and does not fire:

```python
from typing import Self  # direct import instead of from _mytyping

# ...same class definition...
# Result: All checks passed!
```

### Ruff version

```
ruff 0.15.10 (252f76102 2026-04-09)
```

**Comment by MichaReiser** [comment]
This isn't just specific to `PYI034`, this is an issue for all typing exports when using a custom typing module. You can work around this by setting [`typing_modules`](https://docs.astral.sh/ruff/settings/#lint_typing-modules)

**Comment by jab** [comment]
Thanks for the tip! Confirmed that setting `typing-modules = ["_mytyping"]` resolves the false positive for the reproducer above.

```
$ ruff check test_ruff_self3.py --isolated  # no config
error[PYI034]: `__ior__` methods in classes like `Foo` usually return `self` at runtime
  --> test_ruff_self3.py:14:9

$ ruff check test_ruff_self3.py  # with typing-modules = ["_mytyping"] in ruff.toml
All checks passed!
```
